packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -117,10 +125,13 @@ class IconButton extends StatelessWidget {
         maxHeight: size,
         child: new Align(
           alignment: alignment,
-          child: new Icon(
-            size: size,
-            icon: icon,
-            color: currentColor
+          child: new IconTheme.merge(
+            context: context,
+            data: new IconThemeData(
+              size: size,
```

Comment:
It's interesting the pass the size this way.  It's different than using a SizedBox because the Icon centers the icon of the specified size.